### PR TITLE
fix(app-router): restore hash anchors on history traversal

### PR DESCRIPTION
**Pull request by NathanDrake2406**
## What changed
App Router popstate scroll restoration now falls back to hash-anchor scrolling when the traversed history entry has no vinext numeric scroll state.

A focused App Router e2e fixture and regression tests cover vinext-managed hash-only Link entries through back and forward traversal, including decoded non-latin IDs, named-anchor fallback, and #top.

## Why
vinext sets history.scrollRestoration to manual. Hash-only App Router navigations can push entries without __vinext_scrollY, so forward traversal to /path#anchor updated the URL but skipped the browser native anchor scroll and could leave the page at the top.

## Approach
The popstate restoration path keeps saved numeric scroll state as the first-priority behavior. Only entries without vinext scroll state fall back to window.location.hash.

Hash fallback mirrors practical Next.js semantics: decode the fragment when possible, treat #top and empty fragments as top, prefer id lookup, then fall back to name lookup. Malformed encodings are left as-is instead of throwing. The fallback uses explicit auto scroll behavior to match the existing vinext hash-scroll helpers.

## Validation
- Added a failing Playwright regression first, then verified it passed after the fix: PLAYWRIGHT_PROJECT=app-router vp run test:e2e -- tests/e2e/app-router/nextjs-compat/hash-popstate-scroll.spec.ts
- Expanded the same E2E coverage after review for decoded fragments, named anchors, and #top: PLAYWRIGHT_PROJECT=app-router vp run test:e2e -- tests/e2e/app-router/nextjs-compat/hash-popstate-scroll.spec.ts
- Ran adjacent unit coverage: vp test run tests/app-browser-entry.test.ts
- Ran repo checks: vp check
- Rebuilt the vinext package for local e2e validation: vp run vinext#build

## Risks / follow-ups
Numeric back and forward restoration intentionally remains higher priority than hash scrolling, so entries with saved __vinext_scrollY keep existing behavior.

This does not attempt to refactor the existing duplicate hash-scroll helpers between App Router browser entry and next/navigation; the change stays scoped to the App Router popstate gap.

## Next.js references
- App Router records hash scroll intent in focusAndScrollRef: https://github.com/vercel/next.js/blob/canary/packages/next/src/client/components/segment-cache/navigation.ts#L596-L693
- App Router resolves id and name hash targets in layout-router: https://github.com/vercel/next.js/blob/canary/packages/next/src/client/components/layout-router.tsx#L122-L140
- App Router consumes hash scroll after navigation commits: https://github.com/vercel/next.js/blob/canary/packages/next/src/client/components/layout-router.tsx#L306-L313
- App Router hash-scroll e2e coverage: https://github.com/vercel/next.js/blob/canary/test/e2e/app-dir/navigation/navigation.test.ts#L149-L211
- Pages Router hash target semantics for #top, id, name, and decodeURIComponent: https://github.com/vercel/next.js/blob/canary/packages/next/src/shared/lib/router/router.ts#L2323-L2348

### CODE REVIEW

**Comment by NathanDrake2406** [comment]
/bigbonk review this PR. Focus on blocking correctness issues and test gaps.

**Comment by NathanDrake2406** [comment]
/bigbonk review this PR. Focus on blocking correctness issues and whether the tests cover the real behavior and meaningful edge cases.

File: packages/vinext/src/server/app-browser-entry.ts [hunk 1]
```diff
@@ -789,8 +789,38 @@ function restoreHydrationNavigationContext(
...
+      return;
+    }
+
+    const idElement = document.getElementById(fragment);
+    if (idElement) {
+      idElement.scrollIntoView();
+      return;
+    }
+
+    document.getElementsByName(fragment)[0]?.scrollIntoView();
```
[the remark this answers is not in the document]
Comment (NathanDrake2406) [reply packages/vinext/src/server/app-browser-entry.ts]: Updated in 78ce7dbc to pass `{ behavior: "auto" }` explicitly for both the id and name-anchor branches, matching the existing hash-scroll helpers.

[the remark this answers is not in the document]
Comment (NathanDrake2406) [reply packages/vinext/src/server/app-browser-entry.ts]: Kept the helper dedup out of scope, but added E2E coverage in 78ce7dbc for the richer popstate semantics so this behavior is pinned down: decoded fragments and `getElementsByName` fallback now run through the real App Router history traversal path.

File: tests/e2e/app-router/nextjs-compat/hash-popstate-scroll.spec.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    await page.goto(`${BASE}/nextjs-compat/hash-popstate-scroll`);
+    await waitForAppRouterHydration(page);
+    await expect(page.locator("h1")).toHaveText("Hash Popstate Scroll");
+
+    await page.click("#hash-link");
+    await expect(page).toHaveURL(`${BASE}/nextjs-compat/hash-popstate-scroll#content`);
+    await expect(page.locator("#content")).toBeInViewport();
+
+    await page.goBack();
+    await expect(page).toHaveURL(`${BASE}/nextjs-compat/hash-popstate-scroll`);
```
[the remark this answers is not in the document]
Comment (NathanDrake2406) [reply tests/e2e/app-router/nextjs-compat/hash-popstate-scroll.spec.ts:53]: Covered these branches in 78ce7dbc: decoded non-latin id fragments, named-anchor fallback, and `#top` forward traversal now have App Router E2E assertions.
